src/sideshift.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import type {StandardTx, SwapFuncParams} from './checkSwapService'
+
+const {checkSwapService} = require('./checkSwapService.js')
+const js = require('jsonfile')
+const fetch = require('node-fetch')
+
+const confFileName = './config.json'
+const config = js.readFileSync(confFileName)
+
+const SIDESHIFT_CACHE = './cache/XAIRAW.json'
+const PAGE_LIMIT = 500
+
+type SideShiftTransaction = {
+  quoteId: string,
+  depositAddress: {
+    address: string
+  },
+  depositAsset: string,
+  depositMin: number,
+  depositMax: number,
+  settleAddress: {
+    address: string
+  },
+  settleAsset: string,
+  settleAmount: number,
+  createdAt: string
+}
+
+async function doSideshift (swapFuncParams: SwapFuncParams) {
+  return checkSwapService(fetchSideshift,
+    SIDESHIFT_CACHE,
+    'XAI',
+    swapFuncParams)
+}
+
+async function fetchSideshift (swapFuncParams: SwapFuncParams) {
+  if (!swapFuncParams.useCache) {
+    console.log('Fetching Sideshift...')
+  }
+  let diskCache = {txs: []}
+  try {
+    diskCache = js.readFileSync(SIDESHIFT_CACHE)
+  } catch (e) {
+    console.log(e)
+  }
+
+  const newTransactions: StandardTx[] = []
+  let offset = 0
+
+  while (1 && !swapFuncParams.useCache) {
+    try {
+      const url = `https://sideshift.ai/api/v1/affiliate/completedOrders?limit=${PAGE_LIMIT}&offset=${offset}&affiliateId=${config.sideShiftAffiliateId}`
+      const options = {
+        method: 'GET',
+        headers: {
+          'affiliateSecret': `${config.sideShiftAffiliateSecret}` // TODO: figure out how to authenticate with affiliateSecret
+        }
+      }
+
+      const transactions: SideShiftTransaction[] = await fetch(url, options)
+        .then(response => response.json())
+
+      for (const tx of transactions) {
+        const timestamp = new Date(tx.createdAt).getTime() / 1000
+        const xaiTx: StandardTx = {
+          status: 'complete',
+          inputTXID: tx.quoteId,
+          inputAddress: tx.depositAddress.address,
+          inputCurrency: tx.depositAsset.toUpperCase(),
+          inputAmount: tx.depositMin, // TODO: or depositMax?
```

Comment:
`depositMin` and `depositMax` are set on a per-shift (order) basis. It disallows users to shift incorrect amounts, which would become unprofitable or deplete our inventory too fast. Depositing less (or more) would result in error and initiation of refund.

For `order.type = 'fixed'` orders, both `depositMin` and `depositMax` are set to the same value after ensuring that the requested amount is in-range with the limits for input/output asset. 

The endpoint will only return `fixed` orders. So either `depositMin` or `depositMax` is fine.